GVFS/GVFS.FunctionalTests/Tools/GVFSFunctionalTestEnlistment.cs

```diff
@@ -115,7 +115,18 @@ public static string GetUniqueEnlistmentRootWithSpaces()
         public string GetObjectRoot(FileSystemRunner fileSystem)
         {
             IEnumerable<FileSystemInfo> localCacheRootItems = this.LocalCacheRoot.ShouldBeADirectory(fileSystem).WithItems();
-            localCacheRootItems.Count().ShouldEqual(2, "Expected local cache root to contain 2 items. Actual items: " + string.Join(",", localCacheRootItems));
+
+            FileInfo[] files = localCacheRootItems.Where(info => info is FileInfo).Cast<FileInfo>().ToArray();

```

Comment:
Ah I missed that you were filtering down to just files here. You could consider changing line 117 above to `this.LocalCacheRoot.ShouldBeADirectory(fileSystem).WithFiles()` and have that return just `FileInfo`s to begin with.